geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -143,10 +144,19 @@ object SimpleFeatureTypes {
     renamed
   }
 
-  def encodeType(sft: SimpleFeatureType, includeUserData: Boolean = false): String = {
-    val suffix = if (!includeUserData || sft.getUserData.isEmpty) { "" } else {
-      sft.getUserData.entrySet.map(e => s"${e.getKey}='${e.getValue}'").mkString(";", ",", "")
+  def encodeType(sft: SimpleFeatureType): String = {
+    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+    val suffix = if (sft.isGeomesaUserData && !sft.getUserData.isEmpty) {
+      sft.getUserData.entrySet.filter(e => e.getKey.asInstanceOf[String].startsWith("geomesa"))
+        .map(e => s"${e.getKey}='${StringEscapeUtils.escapeJava(e.getValue.toString)}'").mkString(";", ",", "")
     }
+    else if (sft.isAllUserData && !sft.getUserData.isEmpty) {
```

Comment:
probably makes sense to swap the order so that we check for all data before checking for just geomesa data